panoramix/assets/javascripts/explore.js

```diff
@@ -161,26 +162,25 @@ function initExploreView() {
           })
           .popover('show');
 
-        $('#copy_url').tooltip().click(function() {
+        $('#copy_url').tooltip().click(function () {
           var success = copyURLToClipboard(data);
-
           if (success) {
             $(this).attr("data-original-title", "Copied!").tooltip('fixTitle').tooltip('show');
-            window.setTimeout(function() {
-              $shortner.popover('destroy');
-            }, 1200);
+            window.setTimeout(destroyPopover, 1200);
           }
         });
-        $('#close_shortner').click(function() {
+        $('#close_shortner').click(destroyPopover);
+
+        function destroyPopover() {
```

Comment:
some refactoring was more complicated. e.g., this was to prevent function definitions being too nested in one another.